src/test/scala/fte/data/GraphCreator.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ Copyright 2016 Jo Pol
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program. If not, see http://www.gnu.org/licenses/gpl.html dibl
+*/
+package fte.data
+
+import dibl.LinkProps.{ WhiteStart, threadLink }
+import dibl.proto.TilesConfig
+import dibl.{ Diagram, LinkProps, NewPairDiagram, NodeProps, ThreadDiagram }
+import fte.layout.OneFormTorus
+
+object GraphCreator {
+
+  /**
+    * @param urlQuery parameters for: https://d-bl.github.io/GroundForge/tiles?
+    *                 For now the tile must be a checker tile and
+    *                 the patch size must span 3 columns and 2 rows of checker tiles.
+    *                 A simplified ascii-art view of a pattern definition:
+    *                 +----+----+----+
+    *                 |....|....|....|
+    *                 |....|....|....|
+    *                 +----+----+----+
+    *                 |....|XXXX|....|
+    *                 |....|XXXX|....|
+    *                 +----+----+----+
+    * @return The X's in the pattern definition are added to the returned graph.
+    */
+  def fromDiagram(urlQuery: String): Option[Graph] = {
+    val config = TilesConfig(urlQuery)
+
+//    implicit val diagram: Diagram = ThreadDiagram(NewPairDiagram.create(config))
+//    implicit val scale: Int = 2
+    implicit val diagram: Diagram = NewPairDiagram.create(config)
+    implicit val scale: Int = 1
```

Comment:
Switch the comment in the lines above to run for pair diagrams or thread diagrams. But something is mixed up for threads in the following lines:

https://github.com/d-bl/GroundForge/blob/64129c469d8b0f8dbb33a1a8bd34a2426375109a/src/test/scala/fte/data/GraphCreator.scala#L83-L91

Reply:
Github fails to upload the log, even when zipped. So just one of the failing patterns:

```
rose&tileStitch=ctc&patchWidth=12&patchHeight=4&tile=5831,-4-7&shiftColsSW=-2&shiftRowsSW=2&shiftColsSE=2&shiftRowsSE=2
a10,a11,a12,a13,b10,b11,b12,b13,b20,b21,b22,b23,c10,c11,c12,c13,d10,d11,d12,d13,d20,d21,d22,d23
(b22,a10) deltas(1,-1) false, true, false, false, true
(d21,a10) deltas(-1,-1) true, false, false, false, false
(a10,a11) deltas(-1,-1) false, false, true, false, true
(b23,a11) deltas(1,-1) false, false, true, false, false
(a10,a12) deltas(1,-1) false, false, false, true, false
(d23,a12) deltas(-1,-1) false, false, false, true, true
(a11,a13) deltas(1,-1) true, false, false, false, true
(a12,a13) deltas(-1,-1) false, true, false, false, false
(a12,b10) deltas(1,-1) false, true, false, false, true
(d23,b10) deltas(1,-1) false, false, false, false, false
(a13,b11) deltas(1,-1) false, false, true, false, false
(b10,b11) deltas(-1,-1) false, false, true, false, true
(b10,b12) deltas(1,-1) false, false, false, true, false
(d22,b12) deltas(0,-1) false, true, false, true, true
(b11,b13) deltas(1,-1) true, false, false, false, true
(b12,b13) deltas(-1,-1) false, true, false, false, false
(b13,b20) deltas(-1,-1) false, false, false, false, true
(c11,b20) deltas(-1,-1) true, false, false, false, false
(b11,b21) deltas(0,-1) true, false, true, false, false
(b20,b21) deltas(-1,-1) false, false, true, false, true
(b20,b22) deltas(1,-1) false, false, false, true, false
(c13,b22) deltas(-1,-1) false, false, false, true, true
(b21,b23) deltas(1,-1) true, false, false, false, true
(b22,b23) deltas(-1,-1) false, true, false, false, false
(b12,c10) deltas(1,-1) false, true, false, false, true
(d11,c10) deltas(-1,-1) true, false, false, false, false
(b13,c11) deltas(1,-1) false, false, true, false, false
(c10,c11) deltas(-1,-1) false, false, true, false, true
(c10,c12) deltas(1,-1) false, false, false, true, false
(d13,c12) deltas(-1,-1) false, false, false, true, true
(c11,c13) deltas(1,-1) true, false, false, false, true
(c12,c13) deltas(-1,-1) false, true, false, false, false
(a11,d10) deltas(-1,-1) true, false, false, false, false
(b23,d10) deltas(-1,-1) false, false, false, false, true
(b21,d11) deltas(0,-1) true, false, true, false, false
(d10,d11) deltas(-1,-1) false, false, true, false, true
(a13,d12) deltas(-1,-1) false, false, false, true, true
(d10,d12) deltas(1,-1) false, false, false, true, false
(d11,d13) deltas(1,-1) true, false, false, false, true
(d12,d13) deltas(-1,-1) false, true, false, false, false
(c12,d20) deltas(1,-1) false, true, false, false, true
(d13,d20) deltas(1,-1) false, false, false, false, false
(c13,d21) deltas(1,-1) false, false, true, false, false
(d20,d21) deltas(-1,-1) false, false, true, false, true
(d12,d22) deltas(0,-1) false, true, false, true, true
(d20,d22) deltas(1,-1) false, false, false, true, false
(d21,d23) deltas(1,-1) true, false, false, false, true
(d22,d23) deltas(-1,-1) false, true, false, false, false
WRONG column number 6
```